src/iexamine.cpp

```diff
@@ -1399,6 +1399,11 @@ void iexamine::dirtmound(player *p, map *m, int examx, int examy)
 
 void iexamine::aggie_plant(player *p, map *m, int examx, int examy)
 {
+    if ((debug_mode == true) && query_yn(_("Debug plant to maturity?"))) {
```

Comment:
This isn't what debug mode is used for.  Debug mode makes a bunch of stuff print out extra data.
Probably what you want to do is add an entry to the main debug menu like "force plant growth" that does this.